main.py

```diff
@@ -84,19 +85,17 @@ def update_mw(self, mw_id):
         """Update a maintenance window."""
```

Comment:
The modifications to this method do not cover all the problems specified. For instance, patching the following payloads arises the same problem:

```
payload1 = {
            "start": start.strftime(TIME_FMT),
            "end": new_time.strftime(TIME_FMT),
            "items": []
        }
```
and
```
payload1 = {
            "start": start.strftime(TIME_FMT),
            "end": new_time.strftime(TIME_FMT)
        }
```

The solution resides in reusing the same approach used in create_mw. I have tested that adding the following code between lines 88 and 89, solve the issue:

```
try:            
    maintenance = MW.from_dict(data, self.controller)
except ValueError as err:            
    raise BadRequest(f'{err}')
if maintenance is None:
    raise BadRequest('One or more items are invalid')
```

Reply:
Doing it the way you suggested would create a new maintenance instance instead of updating.
The second payload is valid for updating, as you don't need to include all the arguments, just the ones you want to change.